### PR TITLE
feat: map expand button for mobile

### PR DESCRIPTION
**Pull request by Roni750**
# Description
As described on Issue #252, I've implemented a toggle button that will expand the map, for devices smaller than 430px wide.

## screenshots
!isExpanded:
image

isExpanded:
image

### CODE REVIEW

File: package-lock.json
Comment (NoamGaash) [inline package-lock.json:1]: Hi roni, we don't use `npm install` here. Use `yarn` instead. Please create another commit in the same branch and delete the package-lock file. thanks!

File: yarn.lock
Comment (NoamGaash) [inline yarn.lock:1]: please run `git checkout main yarn.lock` to revert this change, and commit it

File: src/pages/singleLineMap/index.tsx [hunk 1]
```diff
@@ -162,36 +181,38 @@ const SingleLineMapPage = () => {
...
-
-          {paths.map((path) => (
-            <Polyline
-              key={path.vehicleRef}
-              pathOptions={{
-                color: getColorByHashString(path.vehicleRef.toString()),
-              }}
-              positions={path.locations.map(({ lat, lon }) => [lat, lon])}
+        <Button type="primary" className="expand-button" shape="circle" onClick={() => setIsExpanded(!isExpanded)} icon={<ExpandAltOutlined />} />
+        <ExpandedContainer>
```
Comment (NoamGaash) [inline src/pages/singleLineMap/index.tsx:185]:
```suggestion
        <ExpandedContainer className={isExpanded ? 'expanded' : 'collapsed'}>
```
will let you use CSS to control the container design

File: src/pages/singleLineMap/index.tsx [hunk 2]
```diff
@@ -40,7 +43,7 @@ const position: Point = {
 const SingleLineMapPage = () => {
   const { search, setSearch } = useContext(SearchContext)
   const { operatorId, lineNumber, timestamp, routes, routeKey } = search
-
+  const [isExpanded, setIsExpanded] = useState<Boolean>(false)
```
Comment (NoamGaash) [inline src/pages/singleLineMap/index.tsx:46]:
```suggestion
  const [isExpanded, setIsExpanded] = useState<Boolean>(false)
  const toggleExpanded = useCallback(()=>setIsExpanded(expanded => !expanded), [])
```
using `toggleExpanded` will let you have the same function reference

**Review by NoamGaash — CHANGES_REQUESTED** [review]
Welcome to the project! Thanks for your contribution, I'm looking forward to merge your PR! :star_struck: 
(don't create another one. this one will be updated once you'll add new commits to your branch)

**Comment by Roni750** [comment]
> Welcome to the project! Thanks for your contribution, I'm looking forward to merge your PR! 🤩 (don't create another one. this one will be updated once you'll add new commits to your branch)

Hey Noam, thank you so much for the guidelines! I'll do as you said and will update here once it's done.

File: yarn.lock
Comment (NoamGaash) [inline yarn.lock:1]: please revert this change

File: src/pages/singleLineMap/index.tsx [hunk 3]
```diff
@@ -162,36 +171,38 @@ const SingleLineMapPage = () => {
...
-          {paths.map((path) => (
-            <Polyline
-              key={path.vehicleRef}
-              pathOptions={{
-                color: getColorByHashString(path.vehicleRef.toString()),
-              }}
-              positions={path.locations.map(({ lat, lon }) => [lat, lon])}
+        <Button type="primary" className="expand-button" shape="circle" onClick={toggleExpanded} icon={<ExpandAltOutlined />} />
+        <div className={`${isExpanded ? 'expanded' : 'collapsed'}`}>
+          <ExpandableMap center={position.loc} zoom={8} scrollWheelZoom={true}>
```
Comment (NoamGaash) [inline src/pages/singleLineMap/index.tsx:176]:
```suggestion
          <ExpandableMap center={position.loc} zoom={8} scrollWheelZoom={true} className={`${isExpanded ? 'expanded' : 'collapsed'}`}>
```

**Review by NoamGaash — APPROVED** [review]
consider removing the `@media screen and (min-width: 430px) {`.
it could look nice on desktop devices as well.

**Pull request merged**